app/controllers/favorites_controller.rb

```diff
@@ -0,0 +1,19 @@
+class FavoritesController < ApplicationController
+  include ActionView::Helpers::TextHelper
+
+  def index
+    @fav_pets = favorites.pets_favorite.keys.map do |petid|
+      Pet.find(petid)
+    end
```

Comment:
Might be an opportunity for refactoring here; feel free to adjust as you see fit.